cloud/linode/node_controller.go

```diff
@@ -129,13 +129,29 @@ func (s *nodeController) handleNode(ctx context.Context, node *v1.Node) error {
 		return nil
 	}
 
+	privIP, ok := node.Annotations[annLinodeNodePrivateIP]
+	if ok && time.Since(lastUpdate) < s.ttl {
+		return nil
+	}
+
 	linode, err := s.instances.lookupLinode(ctx, node)
 	if err != nil {
 		klog.Infof("instance lookup error: %s", err.Error())
 		return err
 	}
 
-	if uuid == linode.HostUUID {
+	// TODO: Once nodebalancer supports other subnets, instead of using linode.IPv4
+	//       which contains public and private ip, use different call which returns
+	//       all ip's configured on the node
+	linodePrivIP := ""
+	for _, ip := range linode.IPv4 {
+		if Options.LinodeNodePrivateSubnet.Contains(*ip) {
+			linodePrivIP = ip.String()
+			break
+		}
+	}
```

Comment:
The Linode Go client already has a method for getting and splitting these values:

https://github.com/linode/linodego/blob/852aace73a7c6b439176e9b361a607296589f37c/instance_ips.go#L77

https://github.com/linode/linodego/blob/852aace73a7c6b439176e9b361a607296589f37c/instance_ips.go#L10-L36

It's what the terraform provider relies on as well:

https://github.com/linode/terraform-provider-linode/blob/a148b9c6e95d552e42f05497900b6fdd0981e61f/linode/instance/flatten.go#L37-L39

If it were me, I'd match what the terraform provider does to get "private_ip_address".

Reply:
+1, We also do something similar on CAPL: https://github.com/linode/cluster-api-provider-linode/blob/main/cloud/services/loadbalancers.go#L122-L155